google/cloud/internal/rest_carrier_test.cc

```diff
@@ -0,0 +1,46 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifdef GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY
+
+#include "google/cloud/internal/rest_carrier.h"
+#include <gmock/gmock.h>
+
+namespace google {
+namespace cloud {
+namespace rest_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace {
+
+using ::testing::Contains;
+using ::testing::ElementsAre;
+using ::testing::Pair;
+
+TEST(RestCarrierTest, SetAttribute) {
+  auto context = RestContext();
+  RestCarrier rest_carrier(context);
+
+  rest_carrier.Set("header1", "test-value");
```

Comment:
nit: why "header1" when there is only one header?

Reply:
Fair. Changed. I'm guessing copy and paste is why....